index.bs

```diff
@@ -1150,10 +1150,11 @@ run the following steps:
 1. Let |attributes| be a new [=/list=].
 1. If |domain| is not null, then run these steps:
     1. If |domain| starts with U+002E (`.`), then return failure.
-    1. If |host| does not equal |domain| and
-        |host| does not end with U+002E (`.`) followed by |domain|,
+    1. Let |parsedDomain| be the result of [=host parser|host parsing=] |domain|.
+    1. If |host| does not equal |parsedDomain| and
+        |host| does not end with U+002E (`.`) followed by |parsedDomain|,
         then return failure.
-    1. Let |encodedDomain| be the result of [=UTF-8 encode|UTF-8 encoding=] |domain|.
+    1. Let |encodedDomain| be the result of [=UTF-8 encode|UTF-8 encoding=] |parsedDomain|.
```

Comment:
Should we maybe use the same encoding to store the host? I worry that this comparison will be incorrect in other parts of the spec now.

Reply:
Actually since `parsedDomain` is the input, we should be fine.